src/framework/components/animation/component.js

```diff
@@ -236,6 +236,12 @@ Object.assign(AnimationComponent.prototype, {
     onAssetChanged: function (asset, attribute, newValue, oldValue) {
         var i;
         if (attribute === 'resource' || attribute === 'resources') {
+            // If the attribute is 'resources', newValue can be an empty array when the
+            // asset is unloaded. Therefore, we should assign null in this case
+            if (newValue && newValue.length === 0 && attribute === 'resources') {

```

Comment:
You probably want to change the order here. If newValue is a `resource` instead of `resources`, accessing length member is probably wrong.

Just change the order to:
```javascript
if (attribute === 'resources' && newValue && newValue.length === 0) {
```

Reply:
I had the string comparison last as I wanted the 'more' expensive comparison last. `new.length` will be undefined with an Object type so `newValue.length === 0` would be false.

It's also inline with the logic in this function: https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/component.js#L248

Happy to change it though, I'm not overly keen on the order but really wanted the string comparison last 😬

Reply:
Javascript strings are interned so the string test is actually a pointer comparison. (i.e. it's fast)

Reply:
> Javascript strings are interned so the string test is actually a pointer comparison. (i.e. it's fast)

I think that depends on the browser?